app/models/invoice.rb

```diff
@@ -14,4 +15,12 @@ def serialize_order
 
     self.data = Invoice::OrderSerializer.new(order).serializable_hash
   end
+
+  def set_active
+    self.status = 'active'
+  end
+
+  def cancel_previous_invoices
+    order.invoices.where.not(id:).update_all(status: 'cancelled')
```

Comment:
_noob question_: not sure to understand this `.not(id:)` ... Would you please like me to understand this?

Reply:
This is a new ruby syntax from 3.1, it's equivalent to `.not(id: id)`